mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1549,13 +1550,27 @@ def evaluate(
         self.model_type = model_type
         self.evaluator_config = evaluator_config
         self.feature_names = dataset.feature_names
-        self.custom_metrics = custom_metrics
+
         self.custom_artifacts = custom_artifacts
         self.y = dataset.labels_data
         self.col_mapping = self.evaluator_config.get("col_mapping")
         self.pos_label = self.evaluator_config.get("pos_label")
         self.sample_weights = self.evaluator_config.get("sample_weights")
 
+        if extra_metrics and custom_metrics:
+            raise MlflowException(
+                "The 'custom_metrics' parameter in mlflow.evaluate is deprecated. Please update "
+                "your code to only use the 'extra_metrics' parameter instead."
+            )
+        if custom_metrics:
+            _logger.warning(
+                "The 'custom_metrics' parameter in mlflow.evaluate is deprecated. "
+                "Please update your code to use the 'extra_metrics' parameter instead."
+            )
```

Comment:
Can we use `FutureWarning` here?

```python
import logging
import warnings

logger = logging.getLogger(__name__)


def future_warning():
    warnings.warn("This features is deprecated", FutureWarning, stacklevel=2)


def logger_warning():
    logger.warning("This feature is deprcated")


for _ in range(3):
    future_warning()
    logger_warning()
```

```
a.py:16: FutureWarning: This features is deprecated
  future_warning()
This feature is deprcated
This feature is deprcated
This feature is deprcated
```

`_logger.warning` prints out the message every time `evaluate` is called with `custom_metrics`, while `FutureWarning` is shown only once.

Reply:
`FutureWarning` is easier to test. We can use `pytest.warns`.